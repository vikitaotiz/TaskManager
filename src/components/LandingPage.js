import { NavLink } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { fetchTasksFromApi } from '../redux/apiCalls';
import { fetchTasks } from '../redux/tasks/tasks';
import TasksChart from './TasksChart';

const LandingPage = () => {
  const dispatch = useDispatch();

  const tasks = useSelector((state) => state.tasks);
  const completedTasks = (tasks.filter((val) => val.completed)).length;
  const incompleteTasks = (tasks.filter((val) => !val.completed)).length;

  useEffect(async () => {
    const data = await fetchTasksFromApi();
    dispatch(fetchTasks(data));
  }, []);

  return (
    <div className="d-flex flex-column">
      <div className="header">
        <NavLink to="/tasks" className="text-decoration-none text-white">
          <div className="header">
            <i className="fas fa-tasks ml-2" />
            {' '}
            All Tasks
            {' '}
            {tasks.length}
            <hr />
            <TasksChart />
          </div>
        </NavLink>
      </div>
      <div className="row">
        <div className="col-lg-6 col-sm-12 sides">
          <NavLink to="/completedtasks" className="text-decoration-none text-white">
            <i className="fas fa-list-alt" />
            {' '}
            Completed Tasks
            {' '}
            {completedTasks}
          </NavLink>
        </div>
        <div className="col-lg-6 col-sm-12 sides">
          <NavLink to="/incompletetasks" className="text-decoration-none text-white">
            <i className="far fa-list-alt" />
            {' '}
            Incomplete Tasks
            {' '}
            {incompleteTasks}
          </NavLink>
        </div>
      </div>
    </div>
  );
};

export default LandingPage;
