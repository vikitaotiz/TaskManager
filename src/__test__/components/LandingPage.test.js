import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/storeConfig';
import LandingPage from '../../components/LandingPage';

it('should render correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <LandingPage />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
