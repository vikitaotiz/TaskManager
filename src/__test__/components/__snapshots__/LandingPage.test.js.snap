// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="header mt-2"
  >
    <a
      aria-current={null}
      className="text-decoration-none text-white"
      href="/tasks"
      onClick={[Function]}
    >
      <i
        className="fas fa-chart-pie"
      />
       
      Task Manager
       
      0
      <hr />
      <div
        className="chart"
      >
        <canvas
          data-testid="canvas"
          height={150}
          onClick={[Function]}
          role="img"
          width={300}
        />
      </div>
    </a>
  </div>
  <i
    className="text-center text-white mt-3"
  >
    Recent Tasks
  </i>
  <div
    className="row m-0 p-0"
  />
  <div
    className="row m-0 p-0"
  >
    <div
      className="col-6 sides"
    >
      <a
        aria-current={null}
        className="text-decoration-none text-white"
        href="/completedtasks"
        onClick={[Function]}
      >
        <i
          className="fas fa-list-alt"
        />
         
        Completed Tasks
         
        0
      </a>
    </div>
    <div
      className="col-6 sides"
    >
      <a
        aria-current={null}
        className="text-decoration-none text-white"
        href="/incompletetasks"
        onClick={[Function]}
      >
        <i
          className="far fa-list-alt"
        />
         
        Pending Tasks
         
        0
      </a>
    </div>
  </div>
</div>
`;
